code/xcvm/evm/src/Interpreter.sol

```diff
@@ -0,0 +1,595 @@
+        require(success, "decode value failed");
+    }
+
+    function _handleSecurity(bytes calldata program, uint64 pos)
+        internal
+        returns (IRouter.BridgeSecurity security, uint64 newPos)
+    {
+        // reading network
+        bool success;
+        int32 value;
+        (success, newPos, value) = ProtobufLib.decode_enum(pos, program);
+        require(success, "decode key failed");
+        security = IRouter.BridgeSecurity(uint32(value));
+    }
+
+    function _handleSalt(bytes calldata program, uint64 pos) internal returns (bytes memory salt, uint64 newPos) {
+        // reading salt
+        bool success;
+        uint64 size;
+        (size, pos) = _getMessageLength(program, pos);
+        pos = _checkField(program, 1, ProtobufLib.WireType.LengthDelimited, pos);
+        (size, pos) = _getMessageLength(program, pos);
+        newPos = pos + size;
+        salt = program[pos: newPos];
+    }
+
+    function _handleProgram(bytes calldata program, uint64 pos)
+        internal
+        returns (bytes memory subProgram, uint64 newPos)
+    {
+        // reading program
+        uint64 size;
+        (size, pos) = _getMessageLength(program, pos);
+        subProgram = program[pos:pos + size];
+        newPos = pos + size;
+    }
+
+    function _handleSpawnParams(bytes calldata program, uint64 pos)
+        internal
+        returns (
+            uint64 newPos,
+            uint256 maxPos,
+            uint256 networkId,
+            IRouter.BridgeSecurity security,
+            bytes memory salt,
+            bytes memory spawnedProgram
+        )
+    {
+        // reading spawn instruction
+        uint64 size;
+        (size, pos) = _getMessageLength(program, pos);
+        maxPos = pos + size;
+
+        pos = _checkField(program, 1, ProtobufLib.WireType.LengthDelimited, pos);
+        (networkId, pos) = _handleNetwork(program, pos);
+
+        pos = _checkField(program, 2, ProtobufLib.WireType.Varint, pos);
+        (security, pos) = _handleSecurity(program, pos);
+
+        // read salt
+        pos = _checkField(program, 3, ProtobufLib.WireType.LengthDelimited, pos);
+        (salt, pos) = _handleSalt(program, pos);
+
+        // read program
+        pos = _checkField(program, 4, ProtobufLib.WireType.LengthDelimited, pos);
+        (spawnedProgram, newPos) = _handleProgram(program, pos);
+    }
+
+    function _handleSpawn(bytes calldata program, uint64 pos) internal returns (uint64 newPos) {
+        (
+            uint64 pos,
+            uint256 maxPos,
+            uint256 networkId,
+            IRouter.BridgeSecurity security,
+            bytes memory salt,
+            bytes memory spawnedProgram
+        ) = _handleSpawnParams(program, pos);
+        address bridgeAddress = IRouter(routerAddress).getBridge(networkId, security);
+        // TODO The fund should be pulled by the Bridge or sent from here??? which is more secure??
```

Comment:
I'd say we transfer to the bridge after having applied the xcvm transfer fees. Let's open a ticket if it's not already done :). In practice, this will be an IBC bridge, hence w'ell be able to call a function to dispatch a packet